core/node/events/miniblock.go

```diff
@@ -91,19 +91,27 @@ func NextMiniblockTimestamp(prevBlockTimestamp *timestamppb.Timestamp) *timestam
 }
 
 type MiniblockInfo struct {
-	Ref         *MiniblockRef
-	headerEvent *ParsedEvent
-	events      []*ParsedEvent
-	Proto       *Miniblock
+	Ref                *MiniblockRef
+	headerEvent        *ParsedEvent
+	useGetterForEvents []*ParsedEvent // Use events(). Getter checks if events have been initialized.
```

Comment:
🤦‍♂️

Reply:
the other option is to make a submodule `events/miniblock-info` and move the private code there.